kitsune/products/models.py

```diff
@@ -164,25 +164,18 @@ def documents(self, user=None, **kwargs):
         query.update(kwargs)
         return Document.objects.visible(**query)
 
-    def get_absolute_url(self):
-        if self.parent is None:
-            return reverse(
-                "products.documents",
-                kwargs={
-                    "product_slug": self.product.slug,
-                    "topic_slug": self.slug,
-                },
-            )
-        else:
+    def get_absolute_url(self, product_slug=None):
+        kwargs = {"topic_slug": self.slug}
+        named_url = "products.topic_documents"
+        if product_slug:
+            kwargs.update({"product_slug": product_slug})
+            named_url = "products.documents"
+
+        if self.parent:
             assert self.parent.parent is None
-            return reverse(
-                "products.subtopics",
-                kwargs={
-                    "product_slug": self.product.slug,
-                    "topic_slug": self.parent.slug,
-                    "subtopic_slug": self.slug,
-                },
-            )
+            kwargs.update({"topic_slug": self.parent.slug, "subtopic_slug": self.slug})
+            named_url = "products.subtopics"
```

Comment:
When the topic has a parent, that path assumes that a `product_slug` has been provided. Should we `assert` that?

Reply:
I moved all the block within the `if product_slug` clause.